en/ros2/px4_sdk.md

```diff
@@ -0,0 +1,336 @@
+
+1. Make sure you have a working [ROS 2 setup](../ros/ros2_comm.md), with _px4_msgs_ in the ROS 2 workspace.
+2. Clone the SDK into the workspace:
+
+   ```shell
+   cd $ros_workspace/src
+   git clone --recursive https://github.com/PX4/px4_sdk.git
+   ```
+
+   :::note
+   To ensure compatibility, use the latest _main_ branches for PX4, _px4_msgs_ and the SDK.
+   See also [here](https://github.com/PX4/px4_sdk#compatibility-with-px4).
+   :::
+
+3. Build the workspace:
+
+   ```shell
+   cd ..
+   source install/setup.bash
+   colcon build
+   ```
+
+4. In a different shell, start PX4 SITL (you can use any model or simulator):
+
+   ```shell
+   cd $px4-autopilot
+   make px4_sitl gazebo-classic
+   ```
+
+5. Run the micro XRCE agent in a new shell (you can keep it running afterward):
+
+   ```shell
+   MicroXRCEAgent udp4 -p 8888
+   ```
+
+6. Start QGroundControl.
+
+   :::note
+   Use QGroundControl Daily, which supports dynamically updating the list of modes.
+   :::
+
+7. Back in the ROS 2 terminal, run one of the example modes:
+
+   ```shell
+   ros2 run example_mode_manual_cpp example_mode_manual
+   ```
+
+   You should get an output like this:
+
+   ```
+   [DEBUG] [example_mode_manual]: Checking message compatibility...
+   [DEBUG] [example_mode_manual]: Subscriber found, continuing
+   [DEBUG] [example_mode_manual]: Publisher found, continuing
+   [DEBUG] [example_mode_manual]: Registering 'My Manual Mode' (arming check: 1, mode: 1, mode executor: 0)
+   [DEBUG] [example_mode_manual]: Subscriber found, continuing
+   [DEBUG] [example_mode_manual]: Publisher found, continuing
+   [DEBUG] [example_mode_manual]: Got RegisterExtComponentReply
+   [DEBUG] [example_mode_manual]: Arming check request (id=1, only printed once)
+   ```
+
+8. On the PX4 shell, you can check that PX4 registered the new mode:
+
+   ```shell
+   commander status
+   ```
+
+   The output should contain:
+
+   ```{5}
+   INFO  [commander] Disarmed
+   INFO  [commander] navigation mode: Position
+   INFO  [commander] user intended navigation mode: Position
+   INFO  [commander] in failsafe: no
+   INFO  [commander] External Mode 1: nav_state: 23, name: My Manual Mode
+   ```
+
+9. At this point you should be able to see the mode in QGroundControl as well:
+
+   ![QGC Modes](../../assets/middleware/ros2/px4_sdk/qgc_modes.png)
+
+10. Select the mode, make sure you have a manual control source (physical or virtual joystick), and arm the vehicle.
+    The mode will then activate, and it should print the following output:
+
+    ```
+    [DEBUG] [example_mode_manual]: Mode 'My Manual Mode' activated
+    ```
+
+11. Now you are ready to create your own mode.
+
+## Mode Tutorial
+
+This section steps through an example for a custom mode class.
+For a complete application, check out the [examples in the repository](https://github.com/PX4/px4_sdk/tree/main/examples/cpp).
+
+```cpp{1,5,7-9,24-31}
+class MyMode : public px4_sdk::ModeBase // [1]
```

Comment:
How do I set that my new mode is hidden in the ground station? 

How do I set that my new mode is advanced, and should only be displayed in an advanced section of the UI?